lib/spack/spack/test/install.py

```diff
@@ -162,11 +135,8 @@ def test_partial_install_delete_prefix_and_stage(install_mockery, mock_fetch, wo
         spack.store.STORE.failure_tracker.clear(s, True)
 
         s.package.set_install_succeed()
-        s.package.stage = MockStage(s.package.stage)
-
         s.package.do_install(restage=True)
         assert rm_prefix_checker.removed
-        assert s.package.stage.test_destroyed
```

Comment:
MockStage is copied to a sub-process and mutated there, so checking `s.package.stage` properties is completely broken.

Reply:
(I passed through here while resolving a conflict with #41373, although I don't think there's any conflicts there, I found this comment confusing)

> MockStage is copied to a sub-process and mutated there, so checking s.package.stage properties is completely broken.

Before this PR, the stage destruction operations we care about for this test were performed in `prepare_for_install` (in the parent process). So this PR created the situation you describe (and ideally resolves it). As far as fragility goes: subprocesses are not a major threat to default-false-and-search-for-something-that-set-it-true-style tests.

> also it tests that the stage is deleted afterwards (which happens in all tests...).

Hopefully not for `--keep-stage` tests

> We're not testing whether the previous stage is destroyed before running a new install.

That's what this check was *intending* to test, and what we'd still like to test - is this a current gap in our tests (i.e. to clarify, are you saying this never tested and still doesn't? or are you saying it didn't and now does?).